### ISSUE TITLE
Android: dismissSoftInput in modal view doesn't work

### ISSUE THREAD

**Issue by luke-hawk** [NativeScript/NativeScript#8515]
**Environment**
 - CLI: 6.1.2
 - Cross-platform modules: 6.1.1
 - Android Runtime: 6.1.1

**Describe the bug**
Calling `utils.ad.dismissSoftInput();` within a modal view doesn't close an open soft keyboard.

**To Reproduce**
1. Add a TextField to a Page
2. Open Page with `showModal `
3. Call `utils.ad.dismissSoftInput()` from a button or tap on layout

**Expected behavior**
Softkeyboard should dismiss.

...cheers for the great work on NativeScript! Love it ❤️

**Comment by NickIliev** [comment]
@luke-hawk indeed the method might not be able to close the software keyboard due to its implementation where we are looking for the wrong view.

As a workaround, try to call the `dismissSoftInput` directly by accessing the TextField that has opened the keyboard in the first place.

TypeScript example

```TypeScript (NativeScript Core)
import { TextField } from "tns-core-modules/ui/text-field";
let tf: TextField;

export function onTfLoaded(args): void {
    tf = args.object as TextField;
}

export function onDismiss(): void {
    tf.dismissSoftInput();
}
```

WHere the onTfLoaded is the callback for the **loaded** event of the field responsible for the open keyboard.

XML (NativeScript Core)
```XML
<TextField row="1" editable="true" loaded="onTfLoaded"></TextField>
```

**Comment by luke-hawk** [comment]
@NickIliev Thanks for your quick response :-) 
I see! This would be a good workaround if I only had one TextField in my modal, but in fact I need to have a whole bunch of fields in the modal and hence I wouldn't be able to know which TextField opened the keyboard in the first place.

Any suggestion for a more generic workaround? I've found [this](https://discourse.nativescript.org/t/unable-to-hide-keyboard/4129/9) solution but it seems to be outdated now. However, it still should be possible to somehow find the current modal view and pass it to `dismissSoftInput`, no?

**Comment by adver89** [comment]
One year without any response. This is how NS looks like. There is no any working part of this framework without workaround.
Everytime I want do something a little bit different than in samples I find out there is no simple way to do this. And in the most cases I find thread like this where is no valuable response or you need to add additional code to do simple thing.
It's already 2 years since I started my journey with NS. And if you are at the beginning better don't use NativeScript.

**Comment by luke-hawk** [comment]
@adver89 I guess you're free to contribute and improve things! Rather than complaining you could think about supporting...

**Comment by adver89** [comment]
Yes, you're right. Also I shouldn't complain because it's a free framework. 
However people should be aware that NS is not mature project yet.

**Comment by CatchABus** [comment]
Closed by #10375
